File: sorts/selection.go
package sorts

//SelectionSort []int
func SelectionSort(arr []int) {
	n := len(arr)
	for i := 0; i < n; i++ {
		min := i
		for j := i; j < n; j++ {
			if arr[j] < arr[min] {
				min = j
			}
		}
		arr[i], arr[min] = arr[min], arr[i]
	}
}

File: sorts/Selection.java
package sorts;

public class Selection {

	public static void sort(int[] arr) {
		int n = arr.length;
		for (int i = 0; i < n; i++) {
			int min = i;
			for (int j = i + 1; j < n; j++) {
				if (arr[j] < arr[min]) {
					min = j;
				}
			}
			int o = arr[i];
			arr[i] = arr[min];
			arr[min] = o;
		}
	}

}
